nifi-nar-bundles/nifi-aws-bundle/nifi-aws-parameter-providers/src/test/java/org/apache/nifi/parameter/aws/TestAwsSecretsManagerParameterProvider.java

```diff
@@ -0,0 +1,199 @@
+                throw new IllegalStateException(e);
+            }
+        });
+        return secretsManager;
+    }
+
+    private List<ParameterGroup> runProviderTest(final AWSSecretsManager secretsManager, final int expectedCount,
+                                                 final ConfigVerificationResult.Outcome expectedOutcome) throws InitializationException {
+
+        final AwsSecretsManagerParameterProvider parameterProvider = getParameterProvider();
+        doReturn(secretsManager).when(parameterProvider).configureClient(any());
+        final MockParameterProviderInitializationContext initContext = new MockParameterProviderInitializationContext("id", "name",
+                new MockComponentLog("providerId", parameterProvider));
+        parameterProvider.initialize(initContext);
+
+        final Map<PropertyDescriptor, String> properties = new HashMap<>();
+        final MockConfigurationContext mockConfigurationContext = new MockConfigurationContext(properties, null);
+
+        List<ParameterGroup> parameterGroups = new ArrayList<>();
+        // Verify parameter fetching
+        if (expectedOutcome == ConfigVerificationResult.Outcome.FAILED) {
+            assertThrows(RuntimeException.class, () -> parameterProvider.fetchParameters(mockConfigurationContext));
+        } else {
+            parameterGroups = parameterProvider.fetchParameters(mockConfigurationContext);
+            final int parameterCount = (int) parameterGroups.stream()
+                    .flatMap(group -> group.getParameters().stream())
+                    .count();
+            assertEquals(expectedCount, parameterCount);
+        }
+
+        // Verify config verification
+        final List<ConfigVerificationResult> results = ((VerifiableParameterProvider) parameterProvider).verify(mockConfigurationContext, initContext.getLogger());
+
+        assertEquals(1, results.size());
+        assertEquals(expectedOutcome, results.get(0).getOutcome());
+
+        return parameterGroups;
+    }
+
+    @Test
+    public void testFetchParametersWithNoSecrets() throws InitializationException {
+        final List<ParameterGroup> expectedGroups = Collections.singletonList(new ParameterGroup("MySecret", Collections.emptyList()));
+        runProviderTest(mockSecretsManager(expectedGroups), 0, ConfigVerificationResult.Outcome.SUCCESSFUL);
+    }
+
+    @Test
+    public void testFetchParameters() throws InitializationException {
+        runProviderTest(mockSecretsManager(mockParameterGroups), 8, ConfigVerificationResult.Outcome.SUCCESSFUL);
+    }
+
+    @Test
+    public void testFetchParametersListFailure() throws InitializationException {
+        when(defaultSecretsManager.listSecrets(any())).thenThrow(new AWSSecretsManagerException("Fake exception"));
+        runProviderTest(defaultSecretsManager, 0, ConfigVerificationResult.Outcome.FAILED);
+    }
+
+    @Test
+    public void testFetchParametersGetSecretFailure() throws InitializationException {
```

Comment:
Recommend moving these Test-annotated methods before all other methods in the class.